openpype/hosts/maya/plugins/publish/extract_camera_alembic.py

```diff
@@ -61,10 +61,25 @@ def process(self, instance):
 
             if bake_to_worldspace:
                 job_str += ' -worldSpace'
-                for member in member_shapes:
-                    self.log.info(f"processing {member}")
+
+                # if baked, drop the camera hierarchy to maintain
+                # clean output and backwards compatibility
+                camera_root = cmds.listRelatives(
+                    camera, parent=True, fullPath=True)[0]
+                job_str += ' -root {0}'.format(camera_root)
+
+                for member in members:
+                    member_content = cmds.listRelatives(
+                        member, ad=True, fullPath=True) or []
+                    # skip hierarchy if it contains only camera
+                    # `member_content` will contain camera + its parents
+                    if camera in member_content \
```

Comment:
Please do not do a lookup per member whether the camera is in member_content while it is a set. This could be very slow, right? Should this be a `set`?

```python
and len(member_content) == len(camera.split("|")) - 2```

What's that trying to solve? This looks like an odd comparison to me.

Reply:
What this is trying to achieve is to allow situations where you have camera inside some hierarchy and  at the same time handle cases where you have hierarchy with camera AND something else.

```
alice_GRP
    |---> bob_GRP
    |        \---> camera1_CAM
    \---> box_GEO
```

in that case you would expect alembic with:

```
camera1_CAM
alice_GRP
    \---> box_GEO
```

but that will fail because `alice_GRP` and `camera1_CAM` have parent relationship.

So what it does is that it gets all descendants of the member and check if the count of them is exactly the same as the path to the camera (minus 2 for the camera transform and shape).

Camera is `|alice_GRP|bob_GRP|camera1_CAM|camera1_CAMShape` and all descendants for member `|alice_GRP` are `["|alice_GRP", "|alice_GRP|bob_GRP"]`

if  count of `member_content` is higher, it means there are more things there than camera.  I think there should be warning logged before `continue`.

> This could be very slow, right?

actually Maya is pretty fast for that one. I've just ran some benchmarks and for 1000 calls I get something aroung  ~0.02 sec

Reply:
Why wouldn't we just check whether this:
```python
descendents = cmds.listRelatives(member, allDescendents=True, fullPath=True) or []
shapes = cmds.ls(descendents, shapes=True, noIntermediate=True, long=True)
cameras = cmds.ls(shapes, type="camera", long=True)
if set(shapes) - set(cameras):
    # There are other shapes than just cameras present
    ...
```

I have a feeling we're overthinking or over-engineering this concept somehow. The fact that this starts behaving differently in a certain edge also feels less consistent than we'd like for a publish. No?

Reply:
damn you are right. Why is it the sets always eludes me even if the are the easiest solution.